bundles/org.openhab.binding.vdr/src/main/java/org/openhab/binding/vdr/internal/VDRHandler.java

```diff
@@ -0,0 +1,347 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.vdr.internal;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPChannel;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPClient;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPClientImpl;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPConnectionException;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPDiskStatus;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPEpgEvent;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPException;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPParseResponseException;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPVolume;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VDRHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Matthias Klocke - Initial contribution
+ */
+@NonNullByDefault
+public class VDRHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(VDRHandler.class);
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    private VDRConfiguration config = new VDRConfiguration();
+
+    private @Nullable ScheduledFuture<?> refreshThreadFuture = null;
+
+    public VDRHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    /**
+     * when disposing refresh thread has to be cancelled
+     */
+    @Override
+    public void dispose() {
+        stopRefreshThread(true);
+    }
+
+    /**
+     * Initialization of {@link VDRHandler}
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(VDRConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // background initialization:
+        scheduler.execute(() -> {
```

Comment:
Any reason that you couldn't just execute `scheduleRefreshThread()` instead of this asynchronous process?

Reply:
Thanks for your advice!
The scheduleRefreshThread did almost everything as the coding in initaliziation. So I changed to your suggestion, but did not yet commit as I have one question about it. 

As I interpret the diagram ([Things | openhab](https://www.openhab.org/docs/concepts/things.html)) the openHAB thing lifecycle allows transitions from initialized back to uninitialized and the back to initialized. So I'd like to keep the stopRefreshThread call in line 93 to ensure that there is no existing refresh thread still running. Would that be OK or is there a better alternative I have overlooked?

Reply:
There is no harm in calling `stopRefreshThread` first if you think it would be safer, although I don't really think it is required.

Reply:
Thanks! 
I left the line in for now but will do some long-term analysis on my local machine with extra debugging to see if and when it's called.

Reply:
Hello @cpmeister !
I have removed the stopRefreshThread  call in initialization now as suggested by you. In my local environment with debug output the method's content was never called in the last month. Therefore I now also think the call is not necessary. 
Apart from that I solved a little bug in the SVDRP client.

Can I do anything else for you?